velox/functions/sparksql/StringToMap.h

```diff
@@ -23,80 +23,111 @@ namespace facebook::velox::functions::sparksql {
 template <typename T>
 struct StringToMapFunction {
   VELOX_DEFINE_FUNCTION_TYPES(T);
-
   // Results refer to strings in the first argument.
   static constexpr int32_t reuse_strings_from_arg = 0;
-
   void call(
       out_type<Map<Varchar, Varchar>>& out,
       const arg_type<Varchar>& input,
       const arg_type<Varchar>& entryDelimiter,
       const arg_type<Varchar>& keyValueDelimiter) {
-    VELOX_USER_CHECK_EQ(
-        entryDelimiter.size(), 1, "entryDelimiter's size should be 1.");
-    VELOX_USER_CHECK_EQ(
-        keyValueDelimiter.size(), 1, "keyValueDelimiter's size should be 1.");
-
-    callImpl(
-        out,
-        toStringView(input),
-        toStringView(entryDelimiter),
-        toStringView(keyValueDelimiter));
+    VELOX_USER_CHECK_GE(
+        entryDelimiter.size(), 1, "entryDelimiter's size should >= 1.");
+    VELOX_USER_CHECK_GE(
+        keyValueDelimiter.size(), 1, "keyValueDelimiter's size should >= 1.");
+    callImpl(out, input, entryDelimiter, keyValueDelimiter);
   }
 
  private:
-  static std::string_view toStringView(const arg_type<Varchar>& input) {
-    return std::string_view(input.data(), input.size());
+  /**
+   * @brief Retrieve the index of the pattern string
+   *
+   * This function retrieves all subscripts in the target string that match the
+   * start and end of the pattern
+   *
+   * @param targetStr The target string to search for
+   * @param pattern Search pattern.
+   */
+  std::vector<std::pair<int, int>> findPatternIndex(
```

Comment:
I find the implementation should be similar with the split function, where empty delimiter, unicode characters should be handled correctly. Would you take a look at the split implementation and check if all cases are considered?

https://github.com/facebookincubator/velox/pull/10248/files#diff-f8a1f11051b531ced5e5c904968040488efc3adef15cd08f805195066e23b0e7R1511-R1571

Reply:
The assertions before the empty string have been filtered out, and Unicode characters can still work properly. I will add a few test samples later

Reply:
> The assertions before the empty string have been filtered out

To reject empty delimiter seems to be not consistent with the behavior of Spark. Could you confirm? I tried below query in Spark and met below exception due to duplicate key for MAP.

> spark.sql("SELECT str_to_map('a:1,b:2,c:3', '', '')").show(false)
org.apache.spark.SparkRuntimeException: [DUPLICATED_MAP_KEY] Duplicate map key : was found, please check the input data. If you want to remove the duplicated keys, you can set "spark.sql.mapKeyDedupPolicy" to "LAST_WIN" so that the key inserted at last takes precedence.

And below query works.

>  spark.sql("SELECT str_to_map('a:1,b:2,c:3', ',', '')").show(false)
+----------------------------+
|str_to_map(a:1,b:2,c:3, ,, )|
+----------------------------+
|{a -> :, b -> :, c -> :}    |
+----------------------------+

If we set the duplicate policy as LAST_WIN as suggested, the first query also works.

> spark.sql("SELECT str_to_map('a:1,b:2,c:3', '', '')").show(false)
+----------------------------------------------------------------------------------------+
|str_to_map(a:1,b:2,c:3, , )                                                             |
+----------------------------------------------------------------------------------------+
|{a -> NULL, : -> NULL, 1 -> NULL, , -> NULL, b -> NULL, 2 -> NULL, c -> NULL, 3 -> NULL}|
+----------------------------------------------------------------------------------------+